File: backend/app/schemas/payment.py
"""Pydantic schemas: Payment (VNPay sandbox thật, task "Quyết định và hoàn
thiện thanh toán") - viết lại HOÀN TOÀN từ placeholder cũ task 8.1
(`PaymentCreateRequest.order_id: str` sai kiểu - `orders.id` là `BigInteger`,
cùng loại lỗi đã sửa cho `ProductRead`/`TopProductRead`, xem
docs/KNOWN_TODOS.md #14).
"""

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel

from app.models.order import PaymentStatus
from app.schemas.base import BaseSchema


class PaymentCreateRequest(BaseModel):
    order_id: int


class PaymentCreateResponse(BaseModel):
    payment_id: int
    order_id: int
    payment_url: str


class PaymentStatusRead(BaseSchema):
    order_id: int
    payment_method: str
    transaction_id: str | None
    amount: Decimal
    status: PaymentStatus
    created_at: datetime
    updated_at: datetime

File: frontend/types/payment.ts
// Khớp CHÍNH XÁC `PaymentCreateRequest`/`PaymentCreateResponse`/`PaymentStatusRead`
// thật (backend/app/schemas/payment.py, task "Quyết định và hoàn thiện
// thanh toán") - thay placeholder cũ task 8.1.

export type PaymentStatus = "pending" | "success" | "failed" | "refunded";

export interface PaymentCreatePayload {
  order_id: number;
}

export interface PaymentCreateResult {
  payment_id: number;
  order_id: number;
  payment_url: string;
}

export interface Payment {
  order_id: number;
  payment_method: string;
  transaction_id: string | null;
  // string, KHÔNG PHẢI number - cùng lý do Order.total_amount/Product.price:
  // Pydantic serialize Decimal thành string.
  amount: string;
  status: PaymentStatus;
  created_at: string;
  updated_at: string;
}
